### ISSUE TITLE
Should pluginWrapper expose the plugin manager?

### ISSUE THREAD

**Issue by jgomer2001** [pf4j/pf4j#353]
Hi,

So we have this method https://www.javadoc.io/static/org.pf4j/pf4j/3.1.0/org.pf4j/org/pf4j/PluginWrapper.html#getPluginManager()

Then I could do

```java
public class EvilPlugin extends Plugin {

    public EvilPlugin (PluginWrapper wrapper) {
        super(wrapper);
        wrapper.getPluginManager().deletePlugin("my enemy");
    }

}
```

and other sort of undesired adversary stuff. I wonder if there is a good reason for exposing the manager...

**Comment by decebals** [comment]
It's a good question.
I think that `PluginManager` was injected in `PluginWrapper` to facilitate the communication between Application (PluginManager) and Plugin. You can read about `push` and `pull` techniques in my [comment](https://github.com/pf4j/pf4j/issues/284#issuecomment-467985538) from #248.
In my projects I prefer to use the `push` approach (a base plugin class for all plugins, and a custom PluginFactory that injects different kind of information on plugin creation). 
What we can do quickly without to break the current API is to externalize the creation of `PluginWrapper` from [AbstractPluginManager#loadPluginFromPath](https://github.com/pf4j/pf4j/blob/master/pf4j/src/main/java/org/pf4j/AbstractPluginManager.java#L821) in a new protected `createPluginWrapper` method. With this modification you can inject in PluginWrapper a kind of `SecurePluginManager` (decorate the real plugin manager - decorator pattern - and disable some sensible/dangerous methods).

**Comment by decebals** [comment]
Closed by #450.
